From ea10031da5c0c7c6d78fffbe05856cfa0f3a7dc2 Mon Sep 17 00:00:00 2001
From: Tam Nguyen <tam.nguyen.xa@renesas.com>
Date: Tue, 27 May 2025 22:09:42 +0700
Subject: [PATCH 33/46] arm64: dts: renesas: ironhide: Add Ethernet support
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Add initial Ethernet port@5 configuration for the RSwitch3
controller on the R-Car X5H (R8A78000) board.

This setup connects port 5 via XPCS to a TI DP83869 PHY in SGMII mode,
with a board-specific MAC address and MDIO configuration. Other ports
(0–4, 6–12) are currently unused and left unconfigured.

Signed-off-by: Tam Nguyen <tam.nguyen.xa@renesas.com>
Signed-off-by: Phong Hoang <phong.hoang.wz@renesas.com>
Signed-off-by: Thanh Quan <thanh.quan.xn@renesas.com>
Signed-off-by: Hai Pham <hai.pham.ud@renesas.com>
---
 .../boot/dts/renesas/r8a78000-ironhide.dts    | 66 +++++++++++++++++++
 1 file changed, 66 insertions(+)

diff --git a/arch/arm64/boot/dts/renesas/r8a78000-ironhide.dts b/arch/arm64/boot/dts/renesas/r8a78000-ironhide.dts
index 366402ee4682..5508cf293f2c 100644
--- a/arch/arm64/boot/dts/renesas/r8a78000-ironhide.dts
+++ b/arch/arm64/boot/dts/renesas/r8a78000-ironhide.dts
@@ -7,6 +7,7 @@
 
 /dts-v1/;
 #include "r8a78000.dtsi"
+#include <dt-bindings/net/ti-dp83869.h>
 
 / {
 	model = "Renesas Ironhide board based on r8a78000";
@@ -146,6 +147,18 @@ &mp_phy {
 };
 
 &pfc {
+	eth25g2_pins: eth25g2 {
+		groups = "eth25g2_mdio", "eth25g2_link", "eth25g2_phyint";
+		function = "eth25g2";
+		drive-strength = <24>;
+	};
+
+	ethes0_pins: ethes0 {
+		groups = "ethes0_match", "ethes0_capture", "ethes0_pps";
+		function = "ethes0";
+		drive-strength = <24>;
+	};
+
 	hscif0_pins: hscif0 {
 		groups = "hscif0_data", "hscif0_ctrl";
 		function = "hscif0";
@@ -160,6 +173,59 @@ i2c1_pins: i2c1 {
 		groups = "i2c1";
 		function = "i2c1";
 	};
+
+	rsw3_pins: rsw3 {
+		groups = "rsw3_match", "rsw3_capture", "rsw3_pps";
+		function = "rsw3";
+		drive-strength = <24>;
+	};
+};
+
+&rsw3_clk {
+	clock-frequency = <66660000>;
+};
+
+&rswitch3 {
+	status = "okay";
+	pinctrl-0 = <&rsw3_pins>, <&eth25g2_pins>, <&ethes0_pins>;
+	pinctrl-names = "default";
+
+	ethernet-ports {
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		// TODO: Add more ports (e.g., port@0, port@5) as needed
+		port@5 {
+			reg = <5>;
+			renesas,etha-mii-chan = <4>;
+			renesas,connect_to_xpcs;
+			phy-handle = <&dp83869_phy>;
+			phy-mode = "sgmii";
+			phys = <&eth_pcs 5>;
+
+			mdio {
+				#address-cells = <1>;
+				#size-cells = <0>;
+				dp83869_phy: ethernet-phy@2 {
+					reg = <2>;
+					ti,sgmii-interface;
+					ti,max-output-impedance;
+					ti,refclk-output-enable;
+					ti,clk-output-sel = <DP83869_CLK_O_SEL_REF_CLK>;
+				};
+			};
+		};
+
+		/* Example stub for future ports */
+		//	port@5 {
+		//	reg = <5>;
+		//	phy-handle = <&mv_phy5>;
+		//	phy-mode = "usxgmii";
+		//	local-mac-address = [xx xx xx xx xx xx];
+		//	phys = <&eth_pcs X>;
+		//	...
+		// };
+	};
 };
 
 &rwdt {
-- 
2.34.1

